### PR TITLE
Support validated H3 checkpoint imports and filter incompatible CivitAI files

### LINKED ISSUE #99: Support verified MiniMax H3 checkpoint imports from CivitAI
### What happened?
The CivitAI browser rejects a Checkpoint whose baseModel is `MiniMax H3`, even though Maestro ships H3 First / Last and Omni runtimes for Pruned and Full checkpoints.

The UI reports:
```
Maestro does not yet have a verified checkpoint-import pipeline for CivitAI base 'MiniMax H3'. The file was not assigned to another model family because that would fail during generation.
```

### Reproduction
1. Open the CivitAI browser and select a MiniMax H3 checkpoint.
2. Attempt to import it as a model.
3. The checkpoint architecture picker has no supported target and blocks the import.

This can also be reproduced without a model download through `GET /api/v1/civitai/checkpoint-architectures?base_model=MiniMax%20H3`.

### Expected behavior
Offer the supported H3 workflows and validate the individual SafeTensor transformer before publishing it or registering a finetune. Preserve the selected checkpoint's QKV layout; tensor dimensions alone cannot distinguish grouped from head-interleaved rows. Validate Full versus Pruned AdaLN dimensions, and honor FL2VA/Ref2VA metadata when present. Unrelated architectures, shards, adapters, and unverified packed formats should remain blocked.

### Environment
Maestro 2.0.1, commit a5dddd4 (current origin/main when investigated); direct Linux/WSL installation, Python 3.11, PyTorch 2.10 CUDA 13, RTX 5090 32 GB. The Pinokio updater was not used for this direct installation.

### Implementation investigation
The exclusion is in `app/services/checkpoint_compatibility.py`: `MiniMax H3` has no allowlisted target/signature. H3 loading is already implemented separately. I am preparing a focused PR for validated single-file imports; this is not a request to bypass the checkpoint safety gate.

### Follow-up: reject incompatible exports before downloading weights
Local implementation now also filters checkpoint search results, versions, and files using supported base families and file formats. H3 INT4, NF4, FP4/NVFP4, W4A8/W4A16, GGUF, and archive exports are excluded when identifiable from filenames or precision metadata. Mixed-version uploads retain their supported files. LoRA filtering is unchanged.

The authenticated download stream validates the SafeTensor header before consuming the remaining weights, then retains the final complete-file validation before publication. The UI suggests a pipeline only for explicit workflow/size filename hints and explains that import creates a new model entry. QKV layout remains an explicit choice.

Regression examples include `minimaxH3INT4Convrot_fl2vaPrunedInt4.safetensors` and `minimaxH3ComfyNativeRef2vaW4a8_v10.safetensors`, which previously reached a generic layout error. Both are now rejected before starting a download.

Validation: 44 model-free regression tests passed, repository cleanliness check and required compile checks passed, React production build passed, and live API checks reject both unsupported filenames. Actual H3 generation with a CivitAI checkpoint has not yet been validated; the planned PR will be a draft. Metadata filtering identifies import candidates and cannot guarantee tensor or runtime compatibility.

### PR DESCRIPTION
**Pull request by anubissbe**
CivitAI MiniMax H3 checkpoints were blocked despite existing H3 runtimes, while unsupported exports could consume a full download before a generic compatibility error. This adds validated Full/Pruned First-Last/Omni imports with explicit QKV layout, checks tensor headers at the start of the download stream, and filters unsupported checkpoint families/files/versions from browsing. H3 packed formats including INT4 and W4A8 remain unsupported and are rejected early. Mixed-version uploads retain supported files.

Imports create a new model definition and preserve the selected checkpoint rather than applying built-in path/layout aliases. Final complete-file validation still runs before publication. LoRA browsing is unchanged.

Validation: 46 regression tests passed; `python scripts/verify_clean_repo.py`, required compileall, and `npm run build` passed. Live API checks reject the reported INT4/W4A8 filenames before starting downloads. Metadata filtering is not a guarantee of runtime compatibility.

A supported CivitAI FP8 checkpoint was downloaded and registered successfully. Full generation remains unverified.

Review fixes: QKV labels no longer associate all ConvRot exports with grouped layout; the guidance identifies the built-in DeepBeepMeep/WanGP interleaved definitions. H3 hides load-time INT8 optimization, sends false explicitly, and rejects API/registration requests enabling the unsupported option before reading weights or starting a download.

Closes #99

### CODE REVIEW

File: ui/src/components/LoraBrowser/ModelDetail.tsx [hunk 1]
```diff
@@ -331,12 +350,28 @@ export function ModelDetail({ model, onBack, kind = 'lora' }: Props) {
...
-                Compatible SafeTensor shapes are verified before the file is installed.
+                A new model entry is created automatically. The tensor header is checked before downloading the weights.
               </p>
-              {checkpointSupportReason && (
+              {isH3Checkpoint && (
+                <div className="mt-3">
+                  <label htmlFor="h3-checkpoint-layout" className="text-[11px] text-text-muted uppercase tracking-wider mb-1 block">H3 checkpoint QKV layout</label>
+                  <select id="h3-checkpoint-layout" value={h3QkvLayout} onChange={e => setH3QkvLayout(e.target.value)} className="w-full bg-bg-tertiary border border-border rounded-lg px-3 py-2 text-sm text-text-primary">
+                    <option value="">Select the publisher's export layout…</option>
+                    <option value="grouped">Grouped Q/K/V (Comfy / ConvRot exports)</option>
```
[the remark this answers is not in the document]
Comment (anubissbe) [reply ui/src/components/LoraBrowser/ModelDetail.tsx]: Fixed in 88f090a. Removed the blanket Comfy/ConvRot = grouped labels in both UI and documentation. The guidance now explicitly notes the built-in DeepBeepMeep/WanGP INT8 ConvRot definitions use interleaved and that other exports can differ; the publisher-informed choice remains explicit.

File: app/services/checkpoint_compatibility.py [hunk 2]
```diff
@@ -97,6 +98,12 @@ def as_dict(self, family: str = "") -> dict:
         CheckpointTarget("krea2_raw", "krea2_raw", "Krea 2 RAW"),
         CheckpointTarget("krea2_turbo", "krea2_turbo", "Krea 2 Turbo"),
     ),
+    "minimax h3": (
+        CheckpointTarget("minimax_h3", "minimax_h3", "H3 First / Last — Pruned"),
+        CheckpointTarget("minimax_h3_full", "minimax_h3_full", "H3 First / Last — Full"),
+        CheckpointTarget("minimax_h3_ref2va", "minimax_h3_ref2va", "H3 Omni — Pruned"),
+        CheckpointTarget("minimax_h3_ref2va_full", "minimax_h3_ref2va_full", "H3 Omni — Full"),
```
[the remark this answers is not in the document]
Comment (anubissbe) [reply app/services/checkpoint_compatibility.py:105]: Fixed in 88f090a. H3 hides Optimize VRAM and explicitly submits auto_quantize=false. The API rejects true before starting a download, registration rejects it before reading weights, and H3 registration options force false. Added regressions for both API and registration behavior. All 46 checkpoint/download-state tests and required checks pass.
